rj_gameplay/rj_gameplay/tactic/nmark_tactic.py

```diff
@@ -0,0 +1,115 @@
+from dataclasses import dataclass
+from typing import List, Optional
+
+import stp.action as action
+import stp.rc as rc
+import stp.tactic as tactic
+import stp.role as role
+
+import rj_gameplay.eval
+import rj_gameplay.skill as skills
+from rj_gameplay.skill import mark
+import stp.skill as skill
+
+import numpy as np
+
+def get_closest_enemies_to_ball(num_enemies: int, world_state: rc.WorldState) -> List[rc.Robot]:
+    ball_pt = world_state.ball.pos
+
+    dist_to_enemies = {
+        np.linalg.norm(ball_pt - robot.pose[0:2]): robot
+        for robot in world_state.their_robots
+    }
+
+    # sort dict keys by dist (shortest first)
+    # return enemies that correspond to n shortest dists
+    return [dist_to_enemies[dist] for dist in sorted(dist_to_enemies.keys())[0:num_enemies]]
+
+class marker_cost(role.CostFn):
+    """Pick mark robots based on dist to the ball point
+    """
+    def __init__(self, enemy_to_mark: rc.Robot=None):
+        self.enemy_to_mark = enemy_to_mark 
+
+    def __call__(
+        self,
+        robot: rc.Robot,
+        prev_result: Optional["RoleResult"],
+        world_state: rc.WorldState,
+    ) -> float:
+
+        # TODO: prevent gameplay crashing w/out this check
+        if robot is None or self.enemy_to_mark is None: 
+            return 0
+        return np.linalg.norm(robot.pose[0:2]-self.enemy_to_mark.pose[0:2])
+
+class NMarkTactic(tactic.ITactic):
+    """Marks the n closest enemies to ball with the closest robots on our team to said enemies.
+    """
+    def __init__(self, n: int):
+        self.num_markers = n
+
+        # create empty mark SkillEntry for each robot
+        self.mark_list = [
+            tactic.SkillEntry(mark.Mark())
+            for i in range(self.num_markers)
+        ]
+
+        # create cost func for each robot
+        self.cost_list = [
+            marker_cost()
+            for _ in self.mark_list
+        ]
+        
+    def compute_props(self):
+        pass
+
+    def create_request(self, **kwargs) -> role.RoleRequest:
+        """Creates a sane default RoleRequest.
+        :return: A list of size 1 of a sane default RoleRequest.
+        """
+        pass
```

Comment:
```suggestion
```

Reply:
should be fixed at the interface level, as discussed on Teams, ignoring for now

Reply:
issue #1665 made